contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/MapRDBPushFilterIntoScan.java

```diff
@@ -184,6 +184,8 @@ protected void doPushFilterIntoBinaryGroupScan(final RelOptRuleCall call,
       return; //no filter pushdown ==> No transformation.
     }
 
+    // Set rowCount in newScanSpec so we do not go and fetch rowCount (an expensive operation) again from MapR DB client.
+    newScanSpec.setRowCount(groupScan.getHBaseScanSpec().getRowCount());
```

Comment:
Change the constructor for BinaryTableGroupScan to also pass in TableStats and add a getter for TableStats. This should be sufficient for passing around the rows.